services/PetFoodService.py

```diff
@@ -0,0 +1,95 @@
+from repositories.PetFoodRepository import PetFoodRepository
+from fastapi import Depends, HTTPException
+from schema.pet_food import PetFoodCreate, PetFoodUpdate, PetFoodGet
+from models.pet_food import PetFood
+from passlib.context import CryptContext
+from fastapi.security import OAuth2PasswordBearer
+
+
+oauth2_scheme = OAuth2PasswordBearer(tokenUrl="token")
+pwd_context = CryptContext(schemes=["bcrypt"], deprecated="auto")
+
+
+class PetFoodService:
+    pet_food_repository: PetFoodRepository
+
+    def __init__(
+        self,
+        pet_food_repository: PetFoodRepository = Depends(),
+    ) -> None:
+        self.pet_food_repository = pet_food_repository
+
+    def get(self, id: int) -> PetFoodGet:
+        pet_food = self.pet_food_repository.get(id=id)
+        if not pet_food:
+            raise HTTPException(status_code=404, detail="해당 사료를 찾을 수 없습니다.")
+        return pet_food
+
+    def get_all_pet_foods(self):
+        pet_foods = self.pet_food_repository.get_all_pet_foods()
+        return pet_foods
+
+    def create(self, pet_food_body: PetFoodCreate) -> PetFood:
+        find_pet_food_by_name = self.pet_food_repository.get_pet_food_by_name(
+            pet_food_body.name
+        )
+        if find_pet_food_by_name:
+            raise HTTPException(status_code=409, detail="해당 사료는 이미 존재합니다.")
+
+        nutrients = self.finding_nutrient(pet_food_body.nutrients)
+        components = self.finding_components(pet_food_body.components)
+
+        return self.pet_food_repository.create(
+            PetFood(name=pet_food_body.name, nutrients=nutrients, components=components)
+        )
+
+    def update(self, id: int, pet_food_body: PetFoodUpdate) -> PetFood:
+        pet_food = self.get(id)
+
+        if pet_food_body.nutrients:
+            nutrients = self.finding_nutrient(pet_food_body.nutrients)
+        else:
+            nutrients = pet_food.nutrients
+
+        if pet_food_body.components:
+            components = self.finding_components(pet_food_body.components)
+        else:
+            components = pet_food.components
+
+        pet_food.name = pet_food_body.name
+        pet_food.nutrients = nutrients
+        pet_food.components = components
```

Comment:
작성하신 의도를 생각해보면 다음처럼 처리해보는건 어떨까요? 제가 잘못 생각했다면 코멘트만 남겨주세요 
```suggestion
        if pet_food_body.nutrients:
            pet_food.nutrients = self.finding_nutrient(pet_food_body.nutrients)

        if pet_food_body.components:
            pet_food.components = self.finding_components(pet_food_body.components)

        pet_food.name = pet_food_body.name
```